src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubListener.cs

```diff
@@ -4,34 +4,74 @@
 using Microsoft.Azure.WebJobs.Host.Listeners;
-using Microsoft.Extensions.Logging;
-
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Extensions.Logging;
+using Microsoft.WindowsAzure.Storage.Blob;
+
 namespace Microsoft.Azure.WebJobs.EventHubs
 {
-    internal sealed class EventHubListener : IListener, IEventProcessorFactory
+    internal sealed class EventHubListener : IListener, IEventProcessorFactory, IScaleMonitorProvider
     {
         private static readonly Dictionary<string, object> EmptyScope = new Dictionary<string, object>();
+        private readonly string _functionId;
+        private readonly string _eventHubName;
+        private readonly string _consumerGroup;
+        private readonly string _connectionString;
+        private readonly string _storageConnectionString;
         private readonly ITriggeredFunctionExecutor _executor;
         private readonly EventProcessorHost _eventProcessorHost;
         private readonly bool _singleDispatch;
         private readonly EventHubOptions _options;
         private readonly ILogger _logger;
         private bool _started;
 
-        public EventHubListener(ITriggeredFunctionExecutor executor, EventProcessorHost eventProcessorHost, bool singleDispatch, EventHubOptions options, ILogger logger)
+        private Lazy<EventHubScaleMonitor> _scaleMonitor;
+
+        public EventHubListener(
+            string functionId,
+            string eventHubName,
+            string consumerGroup,
+            string connectionString,
+            string storageConnectionString,
+            ITriggeredFunctionExecutor executor,
+            EventProcessorHost eventProcessorHost,
+            bool singleDispatch,
+            EventHubOptions options,
+            ILogger logger,
+            CloudBlobContainer blobContainer = null)
         {
+            if (String.IsNullOrEmpty(connectionString))
+            {
+                throw new ArgumentNullException("EventHub Connection String must be provided.");
+            }
+
+            _functionId = functionId;
+            _eventHubName = eventHubName;
+            _consumerGroup = consumerGroup;
+            _connectionString = connectionString;
+            _storageConnectionString = storageConnectionString;
             _executor = executor;
             _eventProcessorHost = eventProcessorHost;
             _singleDispatch = singleDispatch;
             _options = options;
             _logger = logger;
+
+            if (blobContainer == null)
+            {
+                _scaleMonitor = new Lazy<EventHubScaleMonitor>(() => new EventHubScaleMonitor(_functionId, _eventHubName, _consumerGroup, _connectionString, _storageConnectionString, _logger));
+            }
+            else
+            {
+                _scaleMonitor = new Lazy<EventHubScaleMonitor>(() => new EventHubScaleMonitor(_functionId, _eventHubName, _consumerGroup, _connectionString, _storageConnectionString, _logger, blobContainer));
+            }

```

Comment:
I'm not a fan of this. This is only to enable the GetMonitor test in EventHubListenerTests. I have to create a listener and if I don't provide a mock container, it'll throw an exception trying to connect to the fake storage account. So it's a bit silly to pass this through the listener for the test.

Reply:
I think you should be able to remove this if block - just pass the blobContainer down regardless